src/network/networkd-brvlan.c

```
@@ -0,0 +1,337 @@
+        int r;
+        char *p;
+        char *_rvalue = NULL;
+
+        assert(rvalue);
+        assert(vid);
+        assert(vid_end);
+
+        _rvalue = strdupa(rvalue);
+        p = strchr(_rvalue, '-');
+        if (p) {
+                *p = '\0';
+                p++;
+                r = safe_atou(_rvalue, vid);
+                if (r < 0)
+                        return r;
+
+                r = is_vid_valid(*vid);
+                if (r < 0)
+                        return r;
+
+                r = safe_atou(p, vid_end);
+                if (r < 0)
+                        return r;
+
+                r = is_vid_valid(*vid_end);
+        } else {
+                r = safe_atou(_rvalue, vid);
+                if (r < 0)
+                        return r;
+
+                r = is_vid_valid(*vid);
+        }
+        return r;
+}
+
+int config_parse_brvlan_vlan(const char *unit, const char *filename,
+                             unsigned line, const char *section,
+                             unsigned section_line, const char *lvalue,
+                             int ltype, const char *rvalue, void *data,
+                             void *userdata) {
+        Network *network = userdata;
+        int r;
+        unsigned vid = UINT_MAX;
+        unsigned vid_end = UINT_MAX;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = parse_vid_range(rvalue, &vid, &vid_end);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse VLAN, ignoring: %s", rvalue);
+                return r;
+        }
+
+        if (UINT_MAX == vid_end)
+                set_bit(vid++, network->br_vid_bitmap);
+        else {
+                if (vid >= vid_end) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Invalid VLAN range, ignoring %s", rvalue);
+                        return r;
```

you should always pass all metadata you have to the log function, i.e. pass the error number you have to log_syntax(). However, that's completely different from actually returning with the error. For that I'd suggest to eat the error up, and return 0, as that's usually nicer for configuration, to keep compatibility and proceed "as good as we can".